### PR TITLE
EID-1568: Updated readme and added in validity days to the go template

### PR DESCRIPTION
**Pull request by tomrosier**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -35,6 +38,12 @@ make deploy                     # install controller with kubectl
 kubectl delete pod/verify-metadata-controller-controller-manager-0
 ```
 
+To get the project to run and build you need to place it under your `$GOROOT` which is typically set to `~/go/src/` the path for the project should look something like `~/go/src/github.com/alphagov/verify-metadata-controller`.
+
+Once you have dep installed you should be able to run `dep ensure` from the root of the project.
```
Comment (Wynndow) [inline README.md]: What does `dep` bring to the party?

File: pkg/hsm/awscloudhsm/cloudhsm.go [hunk 2]
```diff
@@ -157,10 +157,13 @@ func (c *Client) GenerateAndSignMetadata(request hsm.GenerateMetadataRequest) (s
 		tmpMetadataSigningCertPath,
 		"--output", tmpMetadataOutputPath,
 		"--algorithm", "rsa",
+		"--hsm-metadata-signing-label", request.MetadataSigningKeyLabel,
```
Comment (Wynndow) [inline pkg/hsm/awscloudhsm/cloudhsm.go]: Think you need to drop these lines. The params were changed to have "key-label" as the suffix. They're included just below.

File: README.md (hunk 1 above)
Comment (tomrosier) [reply README.md]: Dep is what is used by kubebuilder to provision the dependancies for Go when the project is initially created. Its just re running the install of the dependancies for a project thats already been created.

File: pkg/hsm/awscloudhsm/cloudhsm.go [hunk 3]
```diff
@@ -161,6 +162,7 @@ func (c *Client) GenerateAndSignMetadata(request hsm.GenerateMetadataRequest) (s
 		"--hsm-saml-signing-key-label", request.SamlSigningKeyLabel,
 		samlSigningOption, tmpSAMLSigningCertPath,
 		"--supplied-saml-encryption-cert-file", tmpSAMLEncryptionCertPath,
+		"--validityDays", request.Data.ValidityDays,
```
Comment (idrop) [inline pkg/hsm/awscloudhsm/cloudhsm.go]: The MetadataGenerator `--validityDays` option is an Integer, I'm guessing picocli will marshall this golang string into an Integer, see https://github.com/alphagov/verify-metadata-controller/blob/a80b88661f2b1d4279bdc8b91d32638bf58f2e96/mdgen/src/main/java/uk/gov/ida/mdgen/MetadataGenerator.java#L122-L123

**Pull request merged**